### ISSUE TITLE
RFC: Make `SubTaskPopup` a child of each drawn mark

### ISSUE THREAD

**Issue by eatyourgreens** [zooniverse/front-end-monorepo#1934]
**Package**
lib-classifier

**Is your feature request related to a problem? Please describe.**
`SubTaskPopup` is currently a child of the `InteractionLayer` component. When we want to open a subtask popup, we have to pass in the active mark and a reference to that mark's DOM node, or client bounding rectangle. This means storing DOM data in the state store, so that it can be passed between drawing tasks, marks and the popup. I think there's also some passing of refs which can be hard to follow.

**Describe the solution you'd like**
If each mark has its own subtask form as a child (see PFE for an example of this in action) then mark properties such as the bounding rectangle can be passed down from the Mark to the SubTaskPopup as props. `SubTaskPopup` uses `React.Portal`, so events fired by the popup will bubble up through the React tree, not the DOM, and can be handled in the Mark. I think this will greatly simplify the code for subtasks, particularly for transcription lines (eg. focus management when the popup opens and closes.)

```jsx
<TranscriptionLine>
  <SubTaskPopup />
</TranscriptionLine>
```

The popup could be added to the generic Mark component. I think that wraps all drawn marks.

**Comment by eatyourgreens** [comment]
After working on the drawing task stories a bit more, this pattern would also allow us to write stories for the drawing tools without having to mock the `InteractionLayer` for subtasks.

**Comment by srallen** [comment]
I think I experimented with this some time ago and I recall that the positioning of the subtask broke when making it a child of the mark. So if anyone else has a look at this, that's something to watch out for.
